app/controllers/concerns/authentication.rb

```diff
@@ -59,8 +57,8 @@ def rbac_allowed?
     return true if ActiveModel::Type::Boolean.new.cast(Settings.disable_rbac)
     return valid_cert_auth? if identity_header.cert_based?
 
-    @rbac_api ||= ::RbacApi.new(raw_identity_header)
-    @rbac_api.check_user
+    set_authenticated_user
```

Comment:
Thinking this further, why this has to be called explicitly when it already has a callsite? Won't that mess up things?

Reply:
Look at how `around_action` calls this method in other places and reorganize things please.

Reply:
Anyway, setting the auth user here can lead to pseudo-authorizations as `current_user` might be tested in future locations even if the user is not authorized to do something...